compose_rl/controllers/actor.py

```diff
@@ -46,9 +46,9 @@ def __init__(
 
         # Set LOCAL_RANK based on Ray GPU allocation
         # ray.get_gpu_ids() is empty if ray is not used. 
-        if len(ray.get_gpu_ids()) > 0:
-            os.environ['LOCAL_RANK'] = '0' if is_cuda_visible_devices_set_by_ray(
-            ) else str(ray.get_gpu_ids()[0])
+        print('ray.get_gpu_ids()', ray.get_gpu_ids(), 'setting local rank to', rank)
+
+        os.environ['LOCAL_RANK'] = str(rank) 
```

Comment:
This may not work even if you `%8`, for instance, let's take the multi node setting:
A: [0,1,2,3,4,5,6,7]
B: [0,1,2,3,4,5,6,7]

If the 8 training GPUs are distributed as such (and the inference GPUs are on all other ranks)
A: [0,1,2,3] <- global rank [0,1,2,3] <- local rank
B: [4,5,6,7] <- global rank [0,1,2,3] <- local rank

Then, rank could be 7 and while it should point to `local_rank=3` on node B, but would instead point to an inference node (which is rank 7 on node A)

Reply:
actually I think for multi-node we should definitely make sure that we are not splitting up the gpus like that. Would also lead to slower communication etc.

Reply:
added comment